File: consumers/power_of_4_golang/activities.go
package main

import (
	"context"
	"os"
	"strconv"
	"time"

	"go.temporal.io/sdk/activity"
)

func PowerOf4(ctx context.Context, a int) (int, error) {
	activityInfo := activity.GetInfo(ctx)
	logger := activity.GetLogger(ctx)

	logger.Info("PowerOf4 activity started", "ActivityID", activityInfo.ActivityID, "input", a)
	waitTimeoutStr := os.Getenv("ACTIVITY_2_WAIT_TIMEOUT")
	waitTimeout, err := strconv.Atoi(waitTimeoutStr)
	if err != nil || waitTimeout == 0 {
		waitTimeout = 5
	}
	time.Sleep(time.Duration(waitTimeout) * time.Second)

	result := a * a * a * a
	logger.Info("PowerOf4 activity completed", "input", a, "result", result)

	return result, nil
}

File: consumers/power_of_2_python/activities.py
import os
from asyncio import sleep

from temporalio import activity

wait_timeout = int(os.getenv("ACTIVITY_1_WAIT_TIMEOUT", "5"))


@activity.defn(name="power_of_2")
async def power_of_2(a: int) -> int:
    await sleep(wait_timeout)
    return a**2
